pkg/pc/control/control_test.go

```diff
@@ -137,3 +137,45 @@ func TestUpdate(t *testing.T) {
 		t.Errorf("Expected to erase old annotation field. Instead we have: %s", pc.Annotations["load_balancer_info"])
 	}
 }
+
+func TestPodClusterFromID(t *testing.T) {
+	testAZ := fields.AvailabilityZone("west-coast")
+	testCN := fields.ClusterName("test")
+	testPodID := types.PodID("pod")
+	selector := labels.Everything().
+		Add(fields.PodIDLabel, labels.EqualsOperator, []string{testPodID.String()}).
+		Add(fields.AvailabilityZoneLabel, labels.EqualsOperator, []string{testAZ.String()}).
+		Add(fields.ClusterNameLabel, labels.EqualsOperator, []string{testCN.String()})
+	session := kptest.NewSession()
+	pcstore := pcstoretest.NewFake()
+
+	pcControllerFromLabels := NewPodCluster(testAZ, testCN, testPodID, pcstore, selector, session)
+	pc, err := pcControllerFromLabels.Create(fields.Annotations{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	pcControllerFromLabels = nil
+
+	pcControllerFromID := NewPodClusterFromID(pc.ID, session, pcstore)
+	retrievedPC, err := pcControllerFromID.Get()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if pc.ID != retrievedPC.ID {
+		t.Errorf("Did not get correct PC back from datastore, expected %s, got %s.\n%v", pc.ID, retrievedPC.ID, retrievedPC)
+	}
+
+	errs := pcControllerFromID.Delete()
+	if len(errs) > 0 {
+		t.Fatalf("%v", errs)
+	}
+
+	notFoundPC, err := pcControllerFromID.Get()
+	if err == nil {
+		t.Errorf("Expected not found error, got none")
```

Comment:
since the error message says you are looking for a not-found, would you like to check that it is `pcstore.NoPodCluster` ?